File: dataset_reduzido/codenet/Java/Code/s749703010.java
import javax.swing.text.Style;
import java.io.*;
import java.util.*;
public class Main {
    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////                                                                                                               /////////
////////                                                                                                               /////////
////////   HHHH        HHHH  EEEEEEEEEEEEE   MMMM          MMMM         OOOOOO             SSSSSSS      EEEEEEEEEEEEE  /////////
////////   HHHH        HHHH  EEEEEEEEEEEEE   MMMMMM      MMMMMM      OOO      OOO        SSSS   SSS     EEEEEEEEEEEEE  /////////
////////   HHHH        HHHH  EEEEE           MMMM MMM  MMM MMMM    OOO          OOO    SSSS       SSS   EEEEE          /////////
////////   HHHH        HHHH  EEEEE           MMMM  MMMMMM  MMMM   OOO            OOO   SSSS             EEEEE          /////////
////////   HHHH        HHHH  EEEEE           MMMM          MMMM  OOO              OOO   SSSSSSS         EEEEE          /////////
////////   HHHHHHHHHHHHHHHH  EEEEEEEEEEE     MMMM          MMMM  OOO              OOO      SSSSSS       EEEEEEEEEEE    /////////
////////   HHHHHHHHHHHHHHHH  EEEEEEEEEEE     MMMM          MMMM  OOO              OOO         SSSSSSS   EEEEEEEEEEE    /////////
////////   HHHH        HHHH  EEEEE           MMMM          MMMM   OOO            OOO              SSSS  EEEEE          /////////
////////   HHHH        HHHH  EEEEE           MMMM          MMMM    OOO          OOO     SSS       SSSS  EEEEE          /////////
////////   HHHH        HHHH  EEEEEEEEEEEEE   MMMM          MMMM      OOO      OOO        SSS    SSSS    EEEEEEEEEEEEE  /////////
////////   HHHH        HHHH  EEEEEEEEEEEEE   MMMM          MMMM         OOOOOO             SSSSSSS      EEEEEEEEEEEEE  /////////
////////                                                                                                               /////////
////////                                                                                                               /////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
    static class Pair{
        int l,r;
        Pair(int a,int b){
            l=a;
            r=b;
        }

        @Override
        public String toString() {
            return l+" "+r;
        }
    }
    static int[][][]memo;
    static int[][]a;
    static int n;
    static int dp(int fliped,int idx,int msk){
        if (msk==(1<<n)-1)return 0;
        if (memo[fliped][idx][msk]!=-1)return memo[fliped][idx][msk];
        int x = Integer.bitCount(msk);
        int ff=0;
        int ans = (int)1e9;
        for (int i =0;i<n;i++){
            if ((msk&(1<<i))==0){
                int nf= x%2;
                if (idx==n||a[fliped][idx]<=a[(nf+(ff%2))%2][i]) {
                    ans = Math.min(ans, ff + dp((nf+(ff%2))%2, i, msk | 1 << i));
                }
                ff++;
            }
            else x--;
        }
        return memo[fliped][idx][msk]=ans;
    }
    public static void main(String[] args) throws IOException, InterruptedException {
        Scanner sc = new Scanner(System.in);
        PrintWriter pw = new PrintWriter(System.out);
        n = sc.nextInt();
        a= new int[2][n];
        for (int i =0;i<n ;i++)a[0][i]=sc.nextInt();
        for (int i =0;i<n ;i++)a[1][i]=sc.nextInt();
        memo = new int[2][n+1][1<<n];
        for (int[][]z:memo)for (int[]x:z)Arrays.fill(x,-1);
        int ans=dp(0,n,0);
        pw.println(ans>=1e8?-1:ans);
        pw.flush();
    }
    static class Scanner {
        StringTokenizer st;
        BufferedReader br;

        public Scanner(FileReader r) {
            br = new BufferedReader(r);
        }

        public Scanner(InputStream s) {
            br = new BufferedReader(new InputStreamReader(s));
        }

        public String next() throws IOException {
            while (st == null || !st.hasMoreTokens())
                st = new StringTokenizer(br.readLine());
            return st.nextToken();
        }

        public int nextInt() throws IOException {
            return Integer.parseInt(next());
        }

        public long nextLong() throws IOException {
            return Long.parseLong(next());
        }

        public String nextLine() throws IOException {
            return br.readLine();
        }

        public double nextDouble() throws IOException {
            String x = next();
            StringBuilder sb = new StringBuilder("0");
            double res = 0, f = 1;
            boolean dec = false, neg = false;
            int start = 0;
            if (x.charAt(0) == '-') {
                neg = true;
                start++;
            }
            for (int i = start; i < x.length(); i++)
                if (x.charAt(i) == '.') {
                    res = Long.parseLong(sb.toString());
                    sb = new StringBuilder("0");
                    dec = true;
                } else {
                    sb.append(x.charAt(i));
                    if (dec)
                        f *= 10;
                }
            res += Long.parseLong(sb.toString()) / f;
            return res * (neg ? -1 : 1);
        }

        public boolean ready() throws IOException {
            return br.ready();
        }
    }

}

File: output/Llama-2-100/codenet/Java/Python/s749703010.py
import java.util
import java.io

class Pair:
    def __init__(self, a, b):
        self.l = a
        self.r = b

    def __str__(self):
        return f"{self.l} {self.r}"

class Scanner:
    def __init__(self, file):
        self.br = file
        self.st = None

    def next(self):
        if self.st is None or not self.st.hasMoreTokens():
            self.st = StringTokenizer(self.br.readLine())
        return self.st.nextToken()

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return long(self.next())

    def nextLine(self):
        return self.br.readLine()

    def nextDouble(self):
        return double(self.next())

    def ready(self):
        return self.br.ready()

class Main:
    def __init__(self):
        self.n = int(input())
        self.a = [list(map(int, input().split())) for _ in range(2)]
        self.memo = [[[0 for _ in range(1 << self.n)] for _ in range(2)] for _ in range(self.n + 1)]

    def dp(self, fliped, idx, msk):
        if msk == (1 << self.n) - 1:
            return 0
        if self.memo[fliped][idx][msk] != -1:
            return self.memo[fliped][idx][msk]
        x = len(bin(msk)) - 1
        ans = 1e9
        for i in range(self.n):
            if not msk & (1 << i):
                nf = x % 2
                if idx == self.n or self.a[fliped][idx] <= self.a[(nf + (fliped % 2)) % 2][i]:
                    ans = min(ans, x + self.dp((nf + (fliped % 2)) % 2, i, msk | (1 << i)))
        return self.memo[fliped][idx][msk] = ans

    def main(self):
        sc = Scanner(input())
        pw = PrintWriter(output())
        ans = self.dp(0, self.n, 0)
        pw.println(ans >= 1e8 ? -1 : ans)
        pw.flush()

if __name__ == "__main__":
    Main().main()
